clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -298,15 +298,10 @@ private MetadataCache handleMetadataResponse(MetadataResponse metadataResponse,
                 if (metadata.isInternal())
                     internalTopics.add(metadata.topic());
                 for (MetadataResponse.PartitionMetadata partitionMetadata : metadata.partitionMetadata()) {
-
-                    Consumer<PartitionInfo> addToPartitions = partitionInfo -> {
-                        int epoch = partitionMetadata.leaderEpoch().orElse(RecordBatch.NO_PARTITION_LEADER_EPOCH);
```

Comment:
Note this was a bug. We were using the new leader epoch even if it was stale and we had taken the `PartitionInfo` from the previous update.

Reply:
Do we have a test that covers this bug too?

Reply:
I believe `testStaleMetadata()` does that